pytest_fixtures/satellite_auth.py

```diff
@@ -0,0 +1,153 @@
+import copy
+
+from nailgun import entities
+from pytest import fixture
+
+from robottelo.config import settings
+from robottelo.constants import AUDIENCE_MAPPER
+from robottelo.constants import GROUP_MEMBERSHIP_MAPPER
+from robottelo.constants import LDAP_ATTR
+from robottelo.constants import LDAP_SERVER_TYPE
+from robottelo.datafactory import gen_string
+from robottelo.rhsso_utils import create_mapper
+from robottelo.rhsso_utils import get_rhsso_client_id
+from robottelo.rhsso_utils import run_command
+
+
+@fixture(scope='function')
+def auth_source(module_org, module_loc):
+    ldap_data = {
+        'ldap_user_name': settings.ldap.username,
+        'ldap_user_passwd': settings.ldap.password,
+        'base_dn': settings.ldap.basedn,
+        'group_base_dn': settings.ldap.grpbasedn,
+        'ldap_hostname': settings.ldap.hostname,
+    }
+    return entities.AuthSourceLDAP(
+        onthefly_register=True,
+        account=ldap_data['ldap_user_name'],
+        account_password=ldap_data['ldap_user_passwd'],
+        base_dn=ldap_data['base_dn'],
+        groups_base=ldap_data['group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login_ad'],
+        server_type=LDAP_SERVER_TYPE['API']['ad'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ldap_data['ldap_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture(scope='function')
+def auth_source_ipa(module_org, module_loc):
+    ipa_data = {
```

Comment:
Something to consider here for future updates, especially after the settings to dynaconf change.

We could flatten these two fixtures into a single fixture function that's indirectly parametrized on LDAP/IPA if we simplified the settings lookup defaults for the IPA options.

This would be simple if the settings instances for auth config data had consistent keys, but were nested under their relevant type, like `settings.ldap.username` and `settings.ipa.username` (instead of `settings.ipa.username_ipa`).

Little bit of logic to add ldap/ipa specific options when the parameter includes them and you've got a single flexible fixture.

Reply:
I like that idea. It makes sense to have it single fixture.